include/rdma/fi_var.h

```diff
@@ -40,6 +40,13 @@
 extern "C" {
 #endif
 
+struct fi_env {
+	const struct fi_provider *provider;
```

Comment:
This is giving apps a hook directly into the provider calls.  I don't think we want to support this.  Just provide the provider name.